doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+(Why: Which syncer should be responsible to retrieve the initial parameter value
+      if they are multiple per parameter?)
+
+A syncer MUST support retrieving the mapped parameters value from the mapped hardware.
+(Why: to retrieve a parameter value at the start of the PF.)
+
+A syncer MUST support setting the mapped parameters value to the mapped hardware.
+(Why: to synchronise hardware on parameter change.)
+
+Syncers MUST retrieve and set the parameters value from the PF core.
+(Why: to be able to synchronise with hardware.)
+
+This API MAY be a packed parameter structure, following the C abi without padding.
+(Note: This is what the reference implementation does.)
+(Why: ???)
+
+## Introspection
+The syncer API SHOULD allow introspection of the mapped parameters.
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
+
+## Plugins
+The PF MUST be able to create syncers.
+(Why: to bind on the corresponding parameters.)
+
+### Definition
+The PF creates syncer using syncer builder.
+
+### Identifier
+All syncer mapping the same hardware SHOULD have their builder regrouped in a syncer library.
+(Why: to be able to link a group of parameters and a given hardware.)
+
+A syncer builder MUST have a unique identifier in its containing syncer library.
+(Why: To uniquely identify the syncer that should bind on parameters.
+      Given that the syncer library has already been specified.)
+
+A syncer library MUST have a unique identifier in the host system.
+(Why: To identify the library associated to parameters.)
+
+Syncer build and syncer library identifiers SHOULD be strings.
+(Why: The reference application does so.)
+
+### Loading
+Syncer library or/and builder MAY be loaded from dynamically linked libraries (called syncer plugins).
+(Why: The reference implementation supports it.)
+
+Such syncer plugins SHOULD have an unique entry point that -when called- should register
+its payload (syncer library/builder) in the provided gatherer.
+(Note: This permit to merge multiple syncer libraries in one dll.)
+(Why: The reference implementation supports it.)
+
+Multiple syncer plugins, may depend on each other. The PF should appropriatly handle the case
+and not fail.
+(Why: The reference implementation supports it.)
+
+## Mapping
+### Definition
+A parameter not bound to a syncer is called a "virtual parameter".
+(Todo: remove if not used in the requirements.)
+
+### Requirements
+**TODO**:
+ - Plugins
+ - association builder <-> parameters
+
+## Sync
+
+Syncer SHOULD synchronise the mapped hardware on parameter change.
+(Why: To always keep synchronise the underlined hardware and the PF parameters.)
+
+Syncer SHOULD retrieve parameter value from the hardware if no value has be set since the PF start.
+(Note: This is usually implemented on PF start, initialize the parameter values with the mapped hardware current state.)
+(Why: To allow introspection of the hardware.)
+
```

Comment:
@dbrageul Could you please clarify ? What to you mean by "blocked" and "monitor" ?

I suppose you are thinking about the "needResync" API, right ?

If by "blocked" you mean that the parameter framework should not hang; then that is a matter to be left to the design, IMHO.

What about:

Syncers MAY report an 'out-of-sync' condition indicating that the hardware parameter values are not (or no longer) reflecting the last values set by the Parameter Framework.
(Why: This can happen when the underlying hardware subsystem crashes/reboots/...)

When a syncer reports an out-of-sync condition, the PF MUST try to resync the hardware values.